storm/src/main/storm/mesos/MesosNimbus.java

```diff
@@ -577,288 +409,265 @@ public void assignSlots(Topologies topologies, Map<String, Collection<WorkerSlot
-          Collections.sort(offerResources, new ResourceRoleComparator());
+        if (!offerResources.isFit(mesosStormConf, topologyDetails, workerPort, !subtractExecutorResources)) {
+          LOG.error(String.format("Unable to launch worker %s. Required cpu: %f, Required mem: %f. Available OfferResources : %s",
+                                  workerHost, requiredCpu, requiredMem, offerResources));
+          continue;
         }
 
-        List<Resource> executorCpuResources = getResourcesScalar(offerResources, executorCpu, "cpus");
-        List<Resource> executorMemResources = getResourcesScalar(offerResources, executorMem, "mem");
-        List<Resource> executorPortsResources = null;
-
-        // Question(ksoundararaj):
-        // Shouldn't we be validating executorCpuResources and executorCpuResources to ensure they aren't
-        // empty or less than what was requested?
-        if (!subtractedExecutorResources) {
-          offerResources = subtractResourcesScalar(offerResources, executorCpu, "cpus");
-          offerResources = subtractResourcesScalar(offerResources, executorMem, "mem");
-          subtractedExecutorResources = true;
-        }
-        List<Resource> workerCpuResources = getResourcesScalar(offerResources, workerCpu, "cpus");
-        offerResources = subtractResourcesScalar(offerResources, workerCpu, "cpus");
-        List<Resource> workerMemResources = getResourcesScalar(offerResources, workerMem, "mem");
-        offerResources = subtractResourcesScalar(offerResources, workerMem, "mem");
-        List<Resource> workerPortsResources = getResourcesRange(offerResources, slot.getPort(), "ports");
-        offerResources = subtractResourcesRange(offerResources, slot.getPort(), "ports");
-
-        // Find port for the logviewer
-        if (!subtractedExecutorResources && MesosCommon.startLogViewer(_conf)) {
-          List<Integer> portList = new ArrayList<>();
-          collectPorts(offerResources, portList, 1);
-          int port = Optional.fromNullable((Number) _conf.get(Config.LOGVIEWER_PORT)).or(8000).intValue();
-          executorPortsResources = getResourcesRange(offerResources, port, "ports");
-          if (!executorPortsResources.isEmpty()) {
-            // Was the port available?
-            extraConfig += String.format(" -c %s=true", MesosCommon.AUTO_START_LOGVIEWER_CONF);
-            offerResources = subtractResourcesRange(offerResources, port, "ports");
-          }
-        }
+        List<Resource> executorCpuResources = new ArrayList<>();
+        List<Resource> executorMemResources = new ArrayList<>();
+        List<Resource> executorPortResources = new ArrayList<>();
+        List<Resource> workerCpuResources = new ArrayList<>();
+        List<Resource> workerMemResources = new ArrayList<>();
+        List<Resource> workerPortResources = new ArrayList<>();
 
-        // Set the MesosSupervisor's storm.local.dir value. By default it is set to "storm-local",
```

Comment:
Please restore these comments.  I don't particularly care whether we use MesosCommon for the lookup, just want the comments that explicate the purpose of this knob.